### ISSUE TITLE
[Feature Request] Implement other git hosting provider

### ISSUE THREAD

**Issue by Zombiefleischer** [oleeskild/obsidian-digital-garden#622]
As the title says, I would like to push to other providers than GitHub like GitLab (in my case a selfhosted GitLab instance), Gitea, etc.

**Comment by chrisl8** [comment]
I am also interested in this. Perhaps I can make a PR to accomplish this if the author is open to the idea?

**Comment by PidgeyL** [comment]
I am also interested in this, as it would allow me to fully self-host this project.
As it may be of interest, I would also like to share this here: I created this project to replace Vercel in the workflow: https://github.com/PidgeyL/digitalgarden-selfhosted
It relies on your node.js project, but uses docker to self-host it, rather than rely on Vercel.

**Comment by SyndicatedPillbug** [comment]
I'd love this as well. GitLab offers better features for maintaining a true site and I'd like to take advantage of that.

**Comment by chrisl8** [comment]
I dug into this a bit myself. The issue is that this program uses a 3rd party plugin that is specifically for interfacing with the GitHub API. That is why this code is able to function without actually installing `git` on the system.

So someone needs to figure out if there is an API for other providers, like CodeBerg and GitLab, and then essentially create a light wrapper around those APIs to mirror the existing library interface allowing the existing code to call whichever one it needs to.

In short, the existence of a library for the GitHub API made that part relatively easy, while other providers are not popular enough to have such libraries yet.

TL;DR: This is non-trivial.

**Comment by ghost** [comment]
Shouldn't this be a bit easier now that forestry.md exists? That's added an extra publishing option in the plugin, which means it should be a little easier to add a third option.

I was thinking that third option should be local git repository which we can then push anywhere we like
